DOCS-8762 Remove Code Analysis Frequently Asked Questions Doc Page (#24769)
DataDog/documentation: content/en/getting_started/code_analysis/_index.md

Reviewer: Remove "You can analyze CI pipelines" because this is about Code Analysis, not CI Visibility

the repository URL is `https://github.com/myorganization/myservice.git`, then,
it associates `myservice` to all files in the repository.

If no repository match is found, Datadog attempts to find a match in the
`path` of the file. If there is a service named `myservice`, and the path is `/path/to/myservice/foo.py`, the file is associated with `myservice` because the service name is part of the path. If two services are present
in the path, the service name the closest to the filename is selected.

### Linking teams to code violations and libraries

Datadog automatically associates the team attached to a service when a code violation or library issue is detected. For example, if the file `domains/ecommerce/apps/myservice/foo.py`
is associated with `myservice`, then the team `myservice` will be associated to any violation
detected in this file.

If no services or teams are found, Datadog uses the `CODEOWNERS` [file][15]
in your repository. The `CODEOWNERS` file determines which team owns a file in your Git provider. 

**Note**: You must accurately map your Git provider teams to your [Datadog teams][16] for this feature to function properly.

> 

Reviewer 2: Can we put this section somewhere else than getting started since it's related to more advanced use of the tool?

Author: @juli1 Would it make sense to add this content with an H3 header like `Link services and teams to code violations` in the [Set up Code Analysis on your repository](https://docs.datadoghq.com/code_analysis/?tab=datadog#set-up-code-analysis-on-your-repository) section? This way, the content is still relevant to the reader and would not exist on this Getting Started page. Thanks!

{{< img src="/getting_started/code_analysis/catalog_service.png" alt="A link to the source code directly in GitHub from a detected library vulnerability" style="width:100%" >}}

### Linking services to code violations and libraries

Datadog associates code violations or libraries with relevant services by using the following mechanisms:

1. [Identifying the code location associated with a service using the Service Catalog.](#identifying-the-code-location-in-the-service-catalog)
2. [Detecting usage patterns of files within additional Datadog products.](#detecting-file-usage-patterns)
3. [Searching for the service name in the file path or repository.](#detecting-service-name-in-paths-and-repository-names)

> If one method succeeds, no further matching attempts are made. Each mapping method is detailed below.

Reviewer 3: ```suggestion
If one method succeeds, no further matching attempts are made. Each mapping method is detailed below.
```
Is this supposed to say "each matching method" ?

Author: In this instance, we're talking about the mapping method, which is the process of associating and linking code violations and libraries with services. With "matching method", that could imply a focus on finding exact matches only, and we have multiple options in case one of them doesn't work out. Updating the first one to be more clear.

#### Identifying the code location in the Service Catalog

> The schema version `v3` and later of the Service Catalog allows you
> to add the mapping of your code location for your service. The `codeLocations`
> section specifies the locations of the code with the repository that
> contains the code and its associated `paths`.

Reviewer 3: ```suggestion
The schema version `v3` and later of the Service Catalog allows you
to add the mapping of your code location for your service. The `codeLocations` section specifies the repository containing the code and its associated paths.
```
Re-wording a bit for clarity

The `paths` attribute is a list of [globs][14]
that should match paths in the repository.

{{< code-block lang="yaml" filename="service.datadog.yaml" collapsible="true" >}}
apiVersion: v3
kind: service
metadata:
  name: my-service
datadog:
  codeLocations:
    - repositoryURL: https://github.com/myorganization/myrepo.git
      paths:
        - /path/to/service/code/**
{{< /code-block >}}

#### Detecting file usage patterns

Datadog detects file usage in additional products such as Error Tracking and associate
files with the runtime service. For example, if a service called `foo` has
a log entry or a stack trace containing a file with a path `/modules/foo/bar.py`,
it associates files `/modules/foo/bar.py` to service `foo`.

#### Detecting service name in paths and repository names

> Datadog detects service name in paths and repository names and associates
> the file with the service if it finds a match.

Reviewer 3: ```suggestion
Datadog detects service names in paths and repository names, and associates the file with the service if a match is found.
```
Re-wording a bit for clarity

For a repository match, if there is a service called `myservice` and
the repository URL is `https://github.com/myorganization/myservice.git`, then,
it associates `myservice` to all files in the repository.

If no repository match is found, Datadog attempts to find a match in the

> `path` of the file. If there is a service `myservice` and the path is
> `/path/to/myservice/foo.py`, it associates the file with `myservice` because
> the name of the service is part of the path. If two services are present

Reviewer 3: ```suggestion
`path` of the file. If there is a service named `myservice`, and the path is `/path/to/myservice/foo.py`, the file is associated with `myservice` because the service name is part of the path. If two services are present
```
Re-wording this one a bit too.

in the path, the service name the closest to the filename is selected.

### Linking teams to code violations and libraries

Datadog automatically associates the team attached to a service when a code violation or library issue is detected. For example, if the file `domains/ecommerce/apps/myservice/foo.py`
is associated with `myservice`, then the team `myservice` will be associated to any violation
detected in this file.

If no services or teams are found, Datadog uses the `CODEOWNERS` [file][15]

> in your repository. The `CODEOWNERS` file determines which team owns a file in your Git provider. You need
> to correctly define the mapping between your Git provider teams and your Datadog teams for this feature to work.

Reviewer 3: ```suggestion
in your repository. The `CODEOWNERS` file determines which team owns a file in your Git provider. 

**Note**: You must accurately map your Git provider teams to your Datadog teams for this feature to function properly.
```

Moving this to a Bolded Note to stand out more and reworded for clarity

[10]: /code_analysis/static_analysis/github_actions/
[11]: /code_analysis/github_pull_requests/#update-an-existing-github-app
[12]: /code_analysis/github_pull_requests
[13]: https://app.datadoghq.com/services 
[14]: https://en.wikipedia.org/wiki/Glob_(programming)

> [15]: https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners

Author: ```suggestion
[15]: https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners
[16]: /account_management/teams/
```
